internal/v4/archive.go

```diff
@@ -162,7 +176,18 @@ func (h *Handler) servePutArchive(id *charm.Reference, w http.ResponseWriter, re
 	if req.ContentLength == -1 {
 		return badRequestf(nil, "Content-Length not specified")
 	}
-	if err := h.addBlobAndEntity(id, req.Body, hash, req.ContentLength); err != nil {
+	promulgatedURL := req.Form.Get("promulgated")
```

Comment:
Why do we have this logic in PUT and not in POST?
Also, why a promulgated bool is not sufficient in this case?
Could you add a comment with answers to the questions above?